### PR TITLE
Improve release instructions

### PR DESCRIPTION
**Pull request by locks**
also tried to make each list item a link to the respective section
of the document

### CODE REVIEW

File: ember-releases.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-- API documentation
-- Release blog post
-- Release page (index, stable, beta, LTS)
-- [Glitch Ember starter](https://emberjs.glitch.me)
+These are, **in this order**:
+
+1. [Guides](#guides)
+1. [API documentation](#api-documentation)
+1. [Release blog post](#release-blog-post)
+1. [Release pageg](#release-pages) (index, stable, beta, LTS)
```
Comment (jaredgalanis) [inline ember-releases.md:10]:
```suggestion
1. [Release page](#release-pages) (index, stable, beta, LTS)
```

File: ember-releases.md [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 When a new version of Ember is released,
 the learning team has to make sure some parts of our infrastructure are adequately updated to account for it.
-These are:
-
-- Guides
-- API documentation
-- Release blog post
-- Release page (index, stable, beta, LTS)
-- [Glitch Ember starter](https://emberjs.glitch.me)
+These are, **in this order**:
```
Comment (ijlee2) [inline ember-releases.md:5]:
```suggestion
**Please update the following in the listed order**:
```

File: ember-releases.md (hunk 1 above)
Comment (ijlee2) [inline ember-releases.md:10]:
```suggestion
1. [Release pages](#release-pages) (index, stable, beta, LTS)
```

**Review by jaredgalanis — APPROVED** [review]
Looks good. One small fix.

File: ember-releases.md (hunk 1 above)
Comment (locks) [reply ember-releases.md]: Saw Isaac's suggestion first 😅

**Pull request merged**